### PR TITLE
enable inspect tests

### PR DESCRIPTION
**Pull request by vrothberg**
A surprisingly big change.  A core problem was that `podman inspect`
allows for passing containers AND images with the default `--type=all`.
This only worked partially as the data was processed in isolation which
caused various issues (e.g., two separate outputs instead of one) but it
also caused issues regarding error handling.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: cmd/podman/inspect/inspect.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	imageEngine     entities.ImageEngine
+	options         entities.InspectOptions
+}
+
+func NewInspector(options entities.InspectOptions) (*Inspector, error) {
+	switch options.Type {
+	case ImageType, ContainerType, AllType:
+		// Valid types.
+	default:
+		return nil, errors.Errorf("invalid type %q is must be %q, %q or %q", ImageType, ContainerType, AllType, options.Type)
```
Comment (TomSweeneyRedHat) [inline cmd/podman/inspect/inspect.go:35]: "is must be" -> "it must be"

File: cmd/podman/inspect/inspect.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	case ContainerType:
+		ctrData, err := i.containerEngine.ContainerInspect(ctx, namesOrIDs, i.options)
+		if err != nil {
+			return err
+		}
+		for i := range ctrData {
+			data = append(data, ctrData[i])
+		}
+	default:
+		return errors.Errorf("invalid type %q is must be %q, %q or %q", ImageType, ContainerType, AllType, i.options.Type)
```
Comment (TomSweeneyRedHat) [inline cmd/podman/inspect/inspect.go:110]: 'is' to 'it' here too.

File: cmd/podman/inspect/inspect.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podman/inspect/inspect.go:35]: Eagle eyes :) Thanks!

**Comment by vrothberg** [comment]
Will make the `Inspector` private. It'll reduce boiler plate.

**Comment by vrothberg** [comment]
@baude @jwhonce @mheon @rhatdan PTAL

**Comment by rhatdan** [comment]
LGTM

File: cmd/podman/inspect/inspect.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	options         entities.InspectOptions
+}
+
+// newInspector creates a new inspector based on the specified options.
+func newInspector(options entities.InspectOptions) (*inspector, error) {
+	switch options.Type {
+	case ImageType, ContainerType, AllType:
+		// Valid types.
+	default:
+		return nil, errors.Errorf("invalid type %q: must be %q, %q or %q", ImageType, ContainerType, AllType, options.Type)
```
Comment (jwhonce) [inline cmd/podman/inspect/inspect.go:60]:
```suggestion
		return nil, errors.Errorf("invalid type %q: must be %q, %q or %q", options.Type, ImageType, ContainerType, AllType)
```

File: cmd/podman/inspect/inspect.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	case ContainerType:
+		ctrData, err := i.containerEngine.ContainerInspect(ctx, namesOrIDs, i.options)
+		if err != nil {
+			return err
+		}
+		for i := range ctrData {
+			data = append(data, ctrData[i])
+		}
+	default:
+		return errors.Errorf("invalid type %q: must be %q, %q or %q", ImageType, ContainerType, AllType, i.options.Type)
```
Comment (jwhonce) [inline cmd/podman/inspect/inspect.go:122]:
```suggestion
		return errors.Errorf("invalid type %q: must be %q, %q or %q", i.options.Type, ImageType, ContainerType, AllType)
```

File: cmd/podman/inspect/inspect.go (hunk 3 above)
Comment (vrothberg) [reply cmd/podman/inspect/inspect.go:60]: Oh man, great catch, thanks!

**Comment by vrothberg** [comment]
Updated. Ready to merge.

**Pull request merged**